flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -901,54 +893,12 @@ public StreamStatusMaintainer getStreamStatusMaintainer() {
         return result;
     }
 
-    private boolean triggerCheckpoint(
+    protected boolean triggerCheckpoint(
```

Comment:
Currently not making it abstract is mainly because `MultipleInputStreamTask`  has override `triggerCheckpointAsync`, if we make ` triggerCheckpoint` abstract, then it has implemented an empty method which it does not use. If we want to also make `MultipleInputStreamTask` do not need to override `triggerCheckpointAsync` directly, we have to make the completable future a parameter to `triggerCheckpoint` and let the triggerCheckpoint to complete the future instead of hide it from `triggerCheckpoint`.

Reply:
I also commit a diff to show the changes if we make `triggerCheckpoint` to be abstract: https://github.com/apache/flink/pull/14820/commits/83de6dda8fab862126c1e91010141828b93fe466